src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -369,7 +379,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     ============================================================
   -->
   <Target Name="_GetRestoreProjectStyle"
-    DependsOnTargets="_GetProjectJsonPath"
+    DependsOnTargets="_GetProjectJsonPath;CollectPackageReferences"
```

Comment:
Our aim should be to make command-line and VS restore exactly the same from a build perspective.